src/test/java/jewellery/inventory/unit/service/SaleServiceTest.java

```diff
@@ -139,4 +151,74 @@ void testCreateSaleProductWillThrowsProductIsPartOfAnotherProduct() {
 
     assertThrows(ProductIsContentException.class, () -> saleService.createSale(saleRequestDto));
   }
+
+  @Test
+  void testReturnProductWillThrowsProductNotSoldException() {
+    UUID productId = product.getId();
+    when(productService.getProduct(any(UUID.class))).thenReturn(product);
+    assertThrows(ProductNotSoldException.class, () -> saleService.returnProduct(productId));
+  }
+
+  @Test
+  void testReturnProductWillThrowsProductIsContentException() {
+    product.setPartOfSale(new Sale());
+    product.setContentOf(new Product());
+    UUID productId = product.getId();
+    when(productService.getProduct(any(UUID.class))).thenReturn(product);
+    assertThrows(ProductIsContentException.class, () -> saleService.returnProduct(productId));
+  }
+
+  @Test
+  void testReturnProductWillThrowsSaleNotFoundException() {
+    product.setPartOfSale(new Sale());
+    UUID productId = product.getId();
+    when(productService.getProduct(any(UUID.class))).thenReturn(product);
+    assertThrows(SaleNotFoundException.class, () -> saleService.returnProduct(productId));
+  }
+
+  @Test
+  void testReturnProductSuccessfully() {
+    product.setPartOfSale(sale);
+    Product productBeforeReturn = product;
+
+    assertEquals(1, sale.getProducts().size());
+    assertNotNull(productBeforeReturn.getPartOfSale());
+
+    when(productService.getProduct(any(UUID.class))).thenReturn(product);
+    when(saleRepository.findById(any(UUID.class))).thenReturn(Optional.of(sale));
+    when(saleService.returnProduct(product.getId())).thenReturn(productReturnResponseDto);
+
+    ProductReturnResponseDto productReturnResponseDto = saleService.returnProduct(product.getId());
+
+    assertEquals(0, sale.getProducts().size());
+    assertNull(productReturnResponseDto.getSaleAfter().getId());
+    assertNotNull(productReturnResponseDto.getReturnedProduct());
+    assertNull(productReturnResponseDto.getReturnedProduct().getPartOfSale());
+    assertNotEquals(
+        productReturnResponseDto.getReturnedProduct().getOwner().getId(),
+        productBeforeReturn.getOwner().getId());
+  }
```

Comment:
if we have `testReturnProductSuccessfullyWithTwoProducts`, there is no need for this test I think.